ballerina/listener_oauth2_provider.bal

```diff
@@ -278,3 +278,10 @@ isolated function validateFromCache(cache:Cache oauth2Cache, string token) retur
     }
     return;
 }
+
+isolated function parseExpClaim(json expClaim) returns int|() {
+    if expClaim is string {
+        return checkpanic int:fromString(expClaim);
```

Comment:
There are a few concerns here.

1. Using `checkpanic` is not recommended. 
2. We should handle the errors correctly.

Reply:
Yes, using `checkpanic` is not recommended, but the current implementation panics in several places due to casting, primarily in the `prepareIntrospectionResponse` method. Ideally, these cases should use the ensureType method, and on error, the introspection response should be considered invalid. As a result, OAuth2 validation should fail since the authorization server does not comply with the RFC specification. I have created a separate issue for this: https://github.com/ballerina-platform/ballerina-library/issues/7231, and we can address it with a new PR, including this checkpanic case.

Reply:
So any changes to be made on this? @MohamedSabthar

Reply:
@MohamedSabthar +1

Reply:
> So any changes to be made on this? @MohamedSabthar
 
Will handle it in a separate PR.